griptape/tasks/base_task.py

```diff
@@ -181,7 +185,18 @@ def after_run(self) -> None:
             )
 
     def can_run(self) -> bool:
-        return self.state == BaseTask.State.PENDING and all(parent.is_finished() for parent in self.parents)
+        # If this Task has been skipped or is not pending, it should not run
+        if self.is_skipped() or not self.is_pending():
+            return False
+
+        # If this Task has parents, and _all_ of them are skipped, it should not run
+        if self.parents and all(parent.is_skipped() for parent in self.parents):
+            return False
+
+        # If _all_ this Task's unskipped parents are finished, it should run
+        unskipped_parents = [parent for parent in self.parents if not parent.is_skipped()]
+
+        return all(parent.is_finished() for parent in unskipped_parents)
```

Comment:
I'm not sure this should be the default behavior. A Task may rely on the output of a parent and not know that it was skipped.

Airflow defaults to [all_success](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/dags.html#trigger-rules) which I think we should follow.

What do you think @vachillo?

```suggestion
        # If this Task has been skipped or is not pending, it should not run
        if self.is_skipped() or not self.is_pending():
            return False

        # If this Task has parents, and _any_ of them are skipped, it should not run
        if self.parents and any(parent.is_skipped() for parent in self.parents):
            self.state = BaseTask.State.SKIPPED
            return False

        return all(parent.is_finished() for parent in self.parents)
```

Reply:
i can see the argument either way, but to me it would seem weird if a task has two parents, one gets skipped, and then task (and all subsequent tasks) get skipped. 

The best "real use case" i can come up for "Task with 2 parents where one can be skipped" is a workflow where one of the parents is augmenting the task with some optional data. I wouldnt want it to be skipped in that case.

> A Task may rely on the output of a parent and not know that it was skipped.

imo a user advanced enough to write a workflow where this is a possibility would make sure their code could handle it. and if not, I think a better developer experience is a failure rather than skipping.

Reply:
Having a non-`all_success` default could lead to some really tricky-to-debug behavior where Tasks are receiving partial inputs, but not necessarily failing. With `all_success` as the default, I think it'd be much more obvious that something is wrong since entire Task branches are not running. 

> The best "real use case" i can come up for "Task with 2 parents where one can be skipped" is a workflow where one of the parents is augmenting the task with some optional data. I wouldnt want it to be skipped in that case.

Totally valid use case, I think it would be well suited for a `all_done` trigger rule. 

There are going to be use-cases for all sorts of trigger rules, and ultimately I think we just need to implement more of them so users can choose their behavior. The important thing here is landing on a sensible default for now.

Should I prioritize implementing those before Branch Task is merged?